### ISSUE TITLE
Parent link not marked as active

### ISSUE THREAD

**Issue by alankirk** [danielharan/semantic-menu#3]
Hi there,

I've just set this up with 1 level of sub nav items and selecting one of the child links it's parent isn't being flagged as an active link.

Thanks again in advance,

Alan

**Comment by danielharan** [comment]
It works for me. Can you show me your test case?

**Comment by alankirk** [comment]
Hi,

Seems to work for me now when using named paths in my routes.rb

I was creating each nav item using

level.add page.title, page.permalink

as i didn't want the url to display as <root>/pages/page_name, rather <root>/page_name and then have something in my routes.rb do a redirect.

Thanks for your help anyway, i'll have a think of a workaround for my routes.

**Comment by danielharan** [comment]
Aah - I've not really tried the use case of adding manually. Here the comments block assigns "My Comments" to root's children:

  root.add "comments", "comments_path", :class => 'button' do |comments|
    comments.add "My Comments", "my_comments_path"

You'd have to hack the library to get it to work that way; it's probably much easier if you name the level.

**Comment by alankirk** [comment]
I managed a workaround in my routes.rb file to let me change the named routes

map.page ':path', :action => 'show', :controller => "pages"

On a side note, i wasn't sure how the whole fork/branch thing worked for updating the code. I made a tweak to the "active" links, so that it wraps the "active" text in a strong tag, so that it's still highlighted with CSS disabled.

Updated code is to lib/semantic_menu.rb function to_s on line 24

def to_s
    content_tag :li, link_to((active? ? "<strong>#{@title}</strong>" : @title), @link, @link_opts) + child_output, ({:class => 'active'} if active?)
  end

Hope that helps!
